File: scripts/extract_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

const file = process.argv[2];
if (!file) {
  console.error('Usage: node extract_pdf.js <pdf-path>');
  process.exit(1);
}

const dataBuffer = fs.readFileSync(file);

pdf(dataBuffer).then(function(data) {
  console.log(data.text);
}).catch(err => {
  console.error('ERROR', err);
  process.exit(1);
});

File: scripts/extract_pdf.py
from pdfminer.high_level import extract_text
import sys

if len(sys.argv) < 2:
    print('Usage: python extract_pdf.py <pdf-path>')
    sys.exit(1)

path = sys.argv[1]
try:
    text = extract_text(path)
    print(text)
except Exception as e:
    print('ERROR:', e)
    sys.exit(2)
